### PR TITLE
Todo list/authentication

### PR DESCRIPTION
**Pull request by faizanmunir017**
TodoList complete working app

### CODE REVIEW

**Comment by JavedJutt** [comment]
Preparing PR description...

File: src/App.tsx [hunk 1]
```diff
@@ -1,7 +1,18 @@
...
+import { BrowserRouter as Router, Route, Routes, Link } from "react-router-dom";
 function App() {
-  return <TodoList></TodoList>;
+  return (
+    <Router>
+      <Routes>
+        <Route path="/" element={<LoginForm />} />
+        <Route path="/login" element={<LoginForm />} />
+        <Route path="/signup" element={<SignUpForm />} />
+        <Route path="/todos" element={<TodoList />} />
```
Comment (Sajawal-dev1) [inline src/App.tsx]: Re-Verify this Login Route Shouldnt be simple '/'

File: src/State/taskActions.ts
Comment (Sajawal-dev1) [inline src/State/taskActions.ts:1]: Move This Actions in Seprate function where you can create types from Funtion

**Comment by JavedJutt** [comment]
## PR Reviewer Guide 🔍

Here are some key observations to aid the review process:

<table>
<tr><td>⏱️&nbsp;<strong>Estimated effort to review</strong>: 4 🔵🔵🔵🔵⚪</td></tr>
<tr><td>🧪&nbsp;<strong>No relevant tests</strong></td></tr>
<tr><td>🔒&nbsp;<strong>No security concerns identified</strong></td></tr>
<tr><td>⚡&nbsp;<strong>Recommended focus areas for review</strong><br><br>

<a href='https://github.com/faizanmunir017/TodoList/pull/6/files#diff-604e1bf2dad542e62ba3fb281d88aed2293ab83ba4900683026e9e743847e7d8R63-R63'><strong>Code Cleanup</strong></a><br>The commented console logs should be removed for cleaner code.

<a href='https://github.com/faizanmunir017/TodoList/pull/6/files#diff-28dad59f16f5806e155c8f41cf9b8346e7d58ac6ae85fbb5f1302f3b127ebcccR54-R54'><strong>Debug Code</strong></a><br>Debugging code like console logs should be removed before merging to maintain code cleanliness.

<a href='https://github.com/faizanmunir017/TodoList/pull/6/files#diff-cd4dd3d999d93a3b2232630b121ba175a769daba5b46854872ac6f9d42c8389bR41-R42'><strong>Error Handling</strong></a><br>Error handling in sagas should be more descriptive and possibly include more robust logging or user notification strategies.

<a href='https://github.com/faizanmunir017/TodoList/pull/6/files#diff-651ee9bcd410cb49a2b159fbad0e560aceb7f76c787b354e4da072c092987531R43-R44'><strong>Logging</strong></a><br>Direct console logging in production code should be avoided or replaced with a more sophisticated logging framework.

</td></tr>
</table>

**Comment by JavedJutt** [comment]
## PR Code Suggestions ✨



<table><thead><tr><td>Category</td><td align=left>Suggestion&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </td><td align=center>Score</td></tr><tbody><tr><td rowspan=3><strong>Enhancement</strong></td>
<td>



<details><summary>Ensure navigation to "/todos" occurs after successful login verification</summary>

___


**Ensure that the login form navigates to the "/todos" route only after confirming the <br>login was successful.**

[src/components/LogIn/LoginForm.tsx [45-48]](https://github.com/faizanmunir017/TodoList/pull/6/files#diff-827dc380d87a2220a8fa9ec023deb7d7b410724e222824628d3fd60899ad2efdR45-R48)

```diff
 dispatch(loginUser({ email, password }));
 console.log("Login successfull :");
-// navigate("/todos");
+navigate("/todos");
```                   
<details><summary>Suggestion importance[1-10]: 7</summary>

Why: The suggestion correctly identifies the need to ensure the navigation to "/todos" only occurs after a successful login, which is crucial for proper user flow and security. The suggested code change is directly relevant and enhances the functionality by using the navigate function after login success.


</details></details></td><td align=center>7

</td></tr><tr><td>



<details><summary>Add error handling to the user registration process</summary>

___


**Add error handling for the registration process to provide feedback to the user on <br>failure.**

[src/components/SignUp/signUpForm.tsx [34-35]](https://github.com/faizanmunir017/TodoList/pull/6/files#diff-903fb4b47c0dd99b8b81c25a632bf4f42a16eda65230da8029e8d7f84a0791c9R34-R35)

```diff
-dispatch(registerUser({ email, password }));
-navigate("/login");
+dispatch(registerUser({ email, password }))
+  .then(() => navigate("/login"))
+  .catch(error => console.error("Registration failed:", error));
```                   
<details><summary>Suggestion importance[1-10]: 7</summary>

Why: Adding error handling to the registration process is a significant improvement for user experience and debugging. The suggestion to catch errors and log them enhances the robustness of the registration feature.


</details></details></td><td align=center>7

</td></tr><tr><td>



<details><summary>Implement error handling for the logout process</summary>

___


**Add error handling for the logout process to manage failures gracefully.**

[src/components/Header/Header.tsx [11-12]](https://github.com/faizanmunir017/TodoList/pull/6/files#diff-08f1eeca099377b71b51c70a42d155ddf8d0d6152e3961b53402a482f70c280cR11-R12)

```diff
-dispatch(logoutUser());
-navigate("/login");
+dispatch(logoutUser())
+  .then(() => navigate("/login"))
+  .catch(error => console.error("Logout failed:", error));
```                   
<details><summary>Suggestion importance[1-10]: 7</summary>

Why: Implementing error handling for the logout process is crucial for a robust application, especially to handle cases where the logout might fail due to server issues or network problems. This suggestion enhances the reliability of the logout functionality.

</details></details></td><td align=center>7

</td></tr><tr><td rowspan=1><strong>Maintainability</strong></td>
<td>



<details><summary>Remove unnecessary commented debug code</summary>

___


**Remove commented debug code to clean up the codebase.**

[src/components/Task/TaskList/TaskList.tsx [54]](https://github.com/faizanmunir017/TodoList/pull/6/files#diff-28dad59f16f5806e155c8f41cf9b8346e7d58ac6ae85fbb5f1302f3b127ebcccR54-R54)

```diff
-// console.log("TaskList toggle task Id", task._id);
 
+
```                   
<details><summary>Suggestion importance[1-10]: 3</summary>

Why: Removing commented out debug code is a good practice for maintaining clean code. However, this is a minor enhancement in terms of overall codebase quality improvement.


</details></details></td><td align=center>3

</td></tr></tr></tbody></table>

File: src/components/LogIn/LoginForm.tsx [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+const LoginForm: React.FC = () => {
+  const dispatch = useDispatch();
+  const navigate = useNavigate();
+  const {
+    register,
+    handleSubmit,
+    formState: { errors, isValid },
+  } = useForm<FormData>({
+    resolver: yupResolver(schema),
+    mode: "onChange",
```
Comment (Sajawal-dev1) [inline src/components/LogIn/LoginForm.tsx]: try with mode All 
I dont think so we need to validate on onChange

**Comment by Sajawal-dev1** [comment]
Remove Console logs
Verify Git Ignore 
Change Paths to Absolute
Error Handling

**Pull request closed without merging**